internal/testrunner/runners/system/servicedeployer/compose.go

```diff
@@ -78,16 +86,25 @@ func (d *DockerComposeServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedSer
 
 	serviceName := inCtxt.Name
 	opts := compose.CommandOptions{
-		Env: append(
-			[]string{fmt.Sprintf("%s=%s", serviceLogsDirEnv, outCtxt.Logs.Folder.Local)},
-			d.sv.Env...),
+		Env: append(d.appEnv,
+			append(
+				[]string{fmt.Sprintf("%s=%s", serviceLogsDirEnv, outCtxt.Logs.Folder.Local)},
+				d.sv.Env...,
+			)...,
+		),
 		ExtraArgs: []string{"--build", "-d"},
 	}
 	err = p.Up(opts)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not boot up service using Docker Compose")
 	}
 
+	// Connect service network with stack network (for the purpose of metrics collection)
```

Comment:
Is it safe to move `docker.ConnectToNetwork` before checking if the service is healthy? Do you think that it won't break if the service doesn't boot up in time?

Reply:
From my understanding it makes no difference for the other cases, and in this one is a requirement (healthcheck only passes once enrolled, and needs to be in the same network to be enrolled). Do you think of any specific scenarios where this might be troublesome?

Reply:
Yeah, but the `compose.go` code is generic, so it may affect also other services like kind or docker based. 

You could verify the behavior by simulating a broken Docker service, for instance: a container that immediately fails setup.